lib/BaseController.js

```diff
@@ -125,6 +117,33 @@ module.exports = class BaseController {
     }
   }
 
+  async preprocessImpersonation() {
+    let impersonationEnabled = await this._impersonation_enabled();
+    let impersonateUser = objectPath.get(this.data, 'object.spec.clusterAuth.impersonateUser', 'false');
+    if (impersonateUser === 'false' || (!impersonationEnabled && impersonateUser != 'razeedeploy' && this.namespace != 'razeedeploy')) {
```

Comment:
writing down for my own thoughts:
- `if (impersonateUser === 'false')`: if this statement is true, it means the webhook is disabled and didnt enforce the resource to have a `clusterAuth.impersonateUser`. in that case we want to patch the resource to make it clear we are running all processing as `razeedeploy`
- `if (!impersonationEnabled && impersonateUser != 'razeedeploy' && this.namespace != 'razeedeploy'))`: if this statement is true, it means impersonation is disabled, some non `razeedeploy` user was defined, and it was deployed to a namespace other than `razeedeploy`. basically, when impersonation is disabled, we want any resource deployed to any namespace other than `razeedeploy` to run as the `razeedeploy` user.
- if impersonation is enabled, or the `razeedeploy` user was already defined, or the resource is in the razeedeploy namespace, we dont need to enforce the razeedeploy user on the resource.